init.lua

```diff
@@ -169,61 +169,63 @@ function ranks.update_nametag(player)
 			prefix = ""
 		end
 
-		player:set_nametag_attributes({
-			text = prefix..name,
-		})
+		if player then
+			player:set_nametag_attributes({
+				text = prefix..name,
+			})
+		end
 
 		return true
 	end
 end
 
--- [function] Set player rank
-function ranks.set_rank(player, rank)
-	if type(player) == "string" then
-		player = minetest.get_player_by_name(player)
+-- [function] Set player rank TODO
+function ranks.set_rank(name, rank)
+	if type(name) == "string" then
+		player = minetest.get_player_by_name(name)
 	end
 
 	if registered[rank] then
-		-- Set attribute
-		player:set_attribute("ranks:rank", rank)
+		storage:set_string(name, rank)
+
 		-- Update nametag
-		ranks.update_nametag(player)
+		ranks.update_nametag(name)
 		-- Update privileges
-		ranks.update_privs(player)
+		ranks.update_privs(name)
 
 		return true
 	end
 end
 
 -- [function] Remove rank from player
-function ranks.remove_rank(player)
-	if type(player) == "string" then
-		player = minetest.get_player_by_name(player)
+function ranks.remove_rank(name)
+	if type(name) == "string" then
+		name = minetest.get_player_by_name(name)
```

Comment:
Once again, this should be effectively flipped. See other comments to the same end.

Reply:
But `minetest.get_player_by_name(name)` != `name:get_player_name()`

Reply:
Oh yeah, this one still does need to be flipped. So
```lua
if type(name) ~= "string" then
    name = name:get_player_name()
end
```
and then define a local player variable assigning `minetest.get_player_by_name(name)` just before the `if player then ... end` statement currently on line 206.

Reply:
Okay, done!